Mathlib/Geometry/Manifold/PartitionOfUnity.lean

```diff
@@ -58,9 +59,40 @@ smooth bump function, partition of unity
 
 universe uι uE uH uM uF
 
-open Function Filter Module Set
+open Bundle Function Filter Module Set
 open scoped Topology Manifold ContDiff
 
+section
+
+-- Let `V` be a real vector bundle over a smooth Hausdorff manifold `M`.
+variable {E : Type*} [NormedAddCommGroup E]
+  [NormedSpace ℝ E] [FiniteDimensional ℝ E]
+  {H : Type*} [TopologicalSpace H] {I : ModelWithCorners ℝ E H}
+  {M : Type*} [TopologicalSpace M] [ChartedSpace H M] {x : M} [IsManifold I ∞ M] [T2Space M]
+  {F : Type*} [NormedAddCommGroup F] [NormedSpace ℝ F] (n : WithTop ℕ∞)
+  {V : M → Type*} [TopologicalSpace (TotalSpace F V)]
+  [∀ x, AddCommGroup (V x)] [∀ x, Module ℝ (V x)] [∀ x : M, TopologicalSpace (V x)]
+  [FiberBundle F V] [VectorBundle ℝ F V]
+
+/-- If `ψ: M → ℝ` a smooth bump function and `s` is a section of a smooth vector bundle `V → M`,
+the scalar product `ψ s` is `C^n` if `s` is `C^n` on an open set containing `tsupport ψ`.
+This is a vector bundle analogue of `contMDiff_of_tsupport`: the total space of `V` has no zero,
+but we only consider sections of the form `ψ s`. -/
+lemma contMDiff_section_of_smul_smoothBumpFunction
```

Comment:
I'm not sure I understand why you write this specifically for smooth bump functions: this would be true for any psi which is C^n and satisfying ht', right?

Reply:
You're right (and I realised this, but didn't fix the PR accordingly).